mbed_host_tests/host_tests_runner/host_test_default.py

```diff
@@ -279,11 +285,16 @@ def start_conn_process():
                             self.logger.prn_wrn("stopped to consume events due to %s event"% key)
                             callbacks_consume = False
                             result = self.RESULT_IO_SERIAL
-                            break
+                            event_queue.put(('__exit_event_queue', 0, time()))
                         elif key == '__exit':
                             # This event is sent by DUT, test suite exited
                             self.logger.prn_inf("%s(%s)"% (key, str(value)))
                             callbacks__exit = True
+                            event_queue.put(('__exit_event_queue', 0, time()))
+                        elif key == '__exit_event_queue':
```

Comment:
This isn't true, `__notify_complete` also sets the result, which is the whole reason why this problem is so intermittent. There are multiple ways to get the result from the host test, which leads to different states of the queue at the point where `__exit` is processed.